### DISCUSSION TITLE
Is maintenance like optimize, vaccuum and rewrite API in scope

### DISCUSSION THREAD

**Discussion by ibotty** [apache/iceberg-rust#571] [Q&A] [answered]
Sorry for the incomplete issue, I pressed enter and it submitted the incomplete issue.

I would like to know whether table maintenance commands as described in https://stackoverflow.com/a/77763196 is in scope.

Thanks.

**Comment by liurenjie1024** [comment, marked as answer]
> Sorry for the incomplete issue, I pressed enter and it submitted the incomplete issue.
> 
> 
> 
> I would like to know whether table maintenance commands as described in https://stackoverflow.com/a/77763196 is in scope.
> 
> 
> 
> Thanks.

Hi, @ibotty Thanks for your interest. The goal of this library  is to have same complete features as java/python API, so it will includes transaction API for the optimization commands. 

But the optimization commands rely on a distributed query engine, and the  roadmap is to integrate this library with rust based engines, such as datafusion, polar-rs.
